### ISSUE TITLE
Checking exists() an a query with Count annotations compiles to unsupported SQL. 

### ISSUE THREAD

**Issue by gagtomm1** [microsoft/mssql-django#119]
I've got this query
```
Manager.values('field').annotate(cnt=Count('id')).filter(cnt__gt=1).exists()
```
Trying to execute it using 1.1.2 cause this error
```
[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Each GROUP BY expression must contain at least one column that is not an outer reference. (164) (SQLExecDirectW); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Statement(s) could not be prepared. (8180)
```
The SQL generated by the lib is
```
SELECT TOP 1 (1) AS [a]
FROM [app_model]
GROUP BY [app_model].[field], (1)
HAVING COUNT_BIG([app_model].[id]) > 1
```
I'm not sure why `(1)` is included in the `GROUP BY` clause, I was expecting something like
```
SELECT TOP 1 (1) AS [a]
FROM app_model
GROUP BY app_model.field
HAVING COUNT(app_model.id) > 1;
```

**Comment by marcperrinoptel** [comment]
NB: from what I see on PostgreSQL, it also generates that weird `(1)` in the `GROUP BY`.

```
SELECT (1) AS "a"
FROM "app_model"
GROUP BY "app_model"."field", (1)
HAVING COUNT("app_model"."id") > 1
LIMIT 1
```

which works, but does the same as without the `, (1)`, so I feel like it's just luckily robust...
Along with the existence of that kind of ticket [https://code.djangoproject.com/ticket/24835](https://code.djangoproject.com/ticket/24835), I'd say this is probably a bit of a **Django bug** - without consequences on PostgreSQL, but with consequences for mssql indeed...
Might be worth asking about that / opening a ticket on Django.

**Comment by absci** [comment]
Thanks for the report, I'll take a look at this issue.

**Comment by absci** [comment]
I was able to reproduce this error. It's indeed a Django design. https://github.com/django/django/blob/470708f50d8c13a50770893b8d7181f5218bf3ac/django/db/models/sql/query.py#L585
We should be able to override it and fix the syntax here.

**Comment by marcperrinoptel** [comment]
The extra `{"a": 1}` probably makes sense for this `exists` design (along with the `limit 1`), but I'm curious how/why it's automatically used in the `group by`, I'll take a closer look.

**Comment by marcperrinoptel** [comment]
Yeah I mean, here https://github.com/django/django/blob/6b53114dd862ec97c282fdfdc83579cbd6d1560d/django/db/models/sql/compiler.py#L135 the "raw 1" from the `select` is automatically pushed to the `group by`.
The comments in this `get_group_by` method are interesting; I still feel like hard values / non-column select expressions should not be pushed to the `group by`.

I can maybe open an question on [Django](https://code.djangoproject.com/); I'm sure we'd prefer not to override `get_group_by` or `exists` ourselves...

**Comment by absci** [comment]
> Yeah I mean, here https://github.com/django/django/blob/6b53114dd862ec97c282fdfdc83579cbd6d1560d/django/db/models/sql/compiler.py#L135 the "raw 1" from the `select` is automatically pushed to the `group by`. The comments in this `get_group_by` method are interesting; I still feel like hard values / non-column select expressions should not be pushed to the `group by`.
> 
> I can maybe open an question on [Django](https://code.djangoproject.com/) (EDIT: done as https://code.djangoproject.com/ticket/33655); I'm sure we'd prefer not to override `get_group_by` or `exists` ourselves...

Sure, feel free to submit to Django if you think it's appropriate for them to fix it. I also prefer not to override too much of Django code in the backend.

**Comment by marcperrinoptel** [comment]
> > Yeah I mean, here https://github.com/django/django/blob/6b53114dd862ec97c282fdfdc83579cbd6d1560d/django/db/models/sql/compiler.py#L135 the "raw 1" from the `select` is automatically pushed to the `group by`. The comments in this `get_group_by` method are interesting; I still feel like hard values / non-column select expressions should not be pushed to the `group by`.
> > I can maybe open an question on [Django](https://code.djangoproject.com/) (EDIT: done as https://code.djangoproject.com/ticket/33655); I'm sure we'd prefer not to override `get_group_by` or `exists` ourselves...
> 
> Sure, feel free to submit to Django if you think it's appropriate for them to fix it. I also prefer not to override too much of Django code in the backend.

Yeah, the Django point of view in the present case might very well be "... well, it works!" but hopefully it's clear enough that it works by sheer luck, that a fix is warranted.

**Comment by marcperrinoptel** [comment]
[https://code.djangoproject.com/ticket/33655](https://code.djangoproject.com/ticket/33655) is now closed (the fix will probably hit 4.0.x?), I think we can close this issue too
